### PR TITLE
Feature - Error Handling

### LINKED ISSUE #16: Error Handling
As a user, I'd like to see descriptive error messages, in order to be able to react to occurred exceptions or errors.

---

### :thought_balloon: Hints & Thoughts
- The back-end already sends error messages compliant with the JSON API specification.
- Maybe there also will be warnings or success messages later. It might be nice to have a generic solution, that can display errors, as well as these "not so important" events.

---

## :white_check_mark: Conditions of satisfaction

- [ ] If an API call returns an error, it is displayed to the user.
- [ ] The solution is also usable for errors that occurred in the front-end only (not only restricted to API errors).
- [ ] The message is displayed in a way that the user clearly sees what has happened and whether it was good or bad (e.g. errors should be red, not only a white text message hidden somewhere on the page).

### PR DESCRIPTION
**Pull request by friedow**
This resolves #16.

### CODE REVIEW

File: src/views/CreateType.vue [hunk 1]
```diff
@@ -214,9 +215,17 @@ export default class Types extends Mixins(Translate) {
...
-    ResourceTypes.create(this.newResourceType);
-    this.$router.push({ path: '/types' });
+  public async createResourceType(): Promise<void> {
+    try {
+      await ResourceTypes.create(this.newResourceType);
+      this.$router.push({ path: '/types' });
+    } catch (e) {
+      this.$notifications.create({
+        level: Level.Critical,
+        title: 'No connection to server: Resources type could not be created. Please try again.',
```
Comment (MaximilianV) [inline src/views/CreateType.vue]: Does it always has to be a lost connection?

File: src/components/NotificationPopup.vue [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  @Prop(Boolean)
+  public visible!: boolean;
+
+  private lastUpdate: number = Number.MAX_SAFE_INTEGER;
+  private timer: number = 0;
+
+  get notificationList(): ListEntry[] {
+    const newNotifications = this.$notifications.notifications.filter(
+      (notification) => {
+        if (!notification.timestamp) {
```
Comment (MaximilianV) [inline src/components/NotificationPopup.vue]: https://github.com/bptlab/rembrandt-frontend/blob/cd30cdb3a1847d5816675c2292ecedbb81d041e3/src/plugins/Notification.ts#L50
ensures that you always have a timestamp. And the notification center also assumes that.
So this `if` might be removed.

**Review by MaximilianV — CHANGES_REQUESTED** [review]

File: src/views/CreateType.vue (hunk 1 above)
Comment (friedow) [reply src/views/CreateType.vue]: done

File: src/components/NotificationPopup.vue (hunk 2 above)
Comment (friedow) [reply src/components/NotificationPopup.vue]: done

**Review by MaximilianV — APPROVED** [review]
👍

**Pull request merged**